modules/create_folder/folders.tf

```diff
@@ -0,0 +1,14 @@
+resource "null_resource" "install" {
+ provisioner "local-exec" {
+    command = "pip install yamldirs"
+  }
+}
+
+resource "null_resource" "folders" {
```

Comment:
why not use terraforms local_file module to create folder+files instead that you have here?
https://registry.terraform.io/providers/hashicorp/local/latest/docs/resources/file